### DISCUSSION TITLE
Share your configuration with results

### DISCUSSION THREAD

**Discussion by Lillifee** [Lillifee/raspiCam#28] [Show and tell]
I"m wondering how you use raspiCam. Let's use this discussion to share your setup (raspberry pi model, camera, lens, etc.) with some results and a short explanation. Maybe you want to share your configurations, tips, and tricks.

**Comment by GLOUPY-RnD** [comment]
This project is almost done !

This is a simple case housing a Raspberry Pi, a relay and a LED push button.
Raspberry 3B+ (will change later for the Zero 2 once it's back on stock for a smaller footprint case)
For livestream : Logitech C920 (will change for a fisheye one later)
For Time Lapse : Raspberry Pi HQ Camera with official 6mm lens.

The case is attached to the printer with double sided tape.

I'll add more info later but to summarize :
- A USB webcam is always running with a smooth live feed, homekit compatible (iOS) (Using [this soft](https://github.com/brutella/hkcam))
- The relay boots and shutdowns the Raspberry Pi when I turn on and off the printer.
- At the end of the ribbon is a Raspberry Pi HQ Camera for Time Lapses.
- The button triggers the Time Lapse mode (Using [this soft](https://github.com/Lillifee/raspiCam)) and blinks when active.
- _A switch is inside the machine and is pushed with the printer's head to take a picture. (currently working on it)_
- A second press of the button disables the Time Lapse and turns off the LED.
- All in a tiny little package attached to my printer.

I printed it in pink. How fashionable =D

GLOUPY

![IMG_0441](https://user-images.githubusercontent.com/5571550/215070514-970a3e59-8d12-4145-883d-f2028cf6710e.jpg)

![IMG_0442](https://user-images.githubusercontent.com/5571550/215070508-423aead7-d590-4d14-988f-e901c039d2db.jpg)

![IMG_0443](https://user-images.githubusercontent.com/5571550/215070499-73ecf9ff-1bcb-4487-a40b-7fe95021f845.jpg)

![IMG_0444](https://user-images.githubusercontent.com/5571550/215070485-051d6c33-5bf1-41f5-875c-d6ee95527415.jpg)

![IMG_0445](https://user-images.githubusercontent.com/5571550/215070462-18178b55-0639-41bc-9e04-ba2b358790d4.jpg)

![IMG_0446](https://user-images.githubusercontent.com/5571550/215070493-55e63f55-5f5a-4067-bdb8-3b7cf77a1095.jpg)

![IMG_0447](https://user-images.githubusercontent.com/5571550/215070503-b12f6204-7d40-4bcf-9937-353e8dd08e98.jpg)

![IMG_0448](https://user-images.githubusercontent.com/5571550/215070512-7ef62d21-b054-4bd9-b056-5a04a76419a6.jpg)

**Reply by Lillifee** [reply]
Hey @GLOUPY-RnD,

That's an amazing setup!!!
Thanks a lot for sharing how you use raspiCam!
It's great to see the pictures and the usage, and i like fashionable pink ;) Courage to color!

I was also waiting quite a while to get a zero 2. I registered on [BerryBase makers club](https://www.berrybase.de/). 
They offer the raspberry pis for a normal price, but usually sold out really quickly :) 

Maybe you can share one of your timelapse videos when you finished!
Thanks a lot!

**Comment by james-dmu** [comment]
This is my setup that is using the raspiCam to control it.

This is a Raspberry Pi 12MP Infrared capable camera. It has a switchable IR filter can auto switch between daylight and IR (or be manually locked to one or the other mode).

![20230423_214558](https://user-images.githubusercontent.com/78555500/236935561-8bfa4ea2-feb6-4edc-bd0a-18b43db8a10f.jpg)

![20230508_214319](https://user-images.githubusercontent.com/78555500/236935503-688c7556-252e-47a9-abfa-ea5d733ea50b.jpg)

![2023-04-23T20-43-08Z](https://user-images.githubusercontent.com/78555500/236935602-2b6f23b9-a3b2-4256-b0b4-d2e33433d731.jpg)

The system is made up of the following.

A Raspberry Pi Zero 2 W.

An Arducam 12.3MP IR-CUT HQ Camera with 6mm CS Lens.

An Arducam CSI to HDMI adapter (this allows me to use the camera a decent distance from the Pi using a HDMI cable to carry the signals).

An LED IR light array (or 2 if need be) that run off 12v.

A self built power box with outputs for the IR lights and a buck converter to convert 12v from a portable battery to 5.1v for the Pi. This box also has a USB connector that was to be used to control a relay inside to turn IR lights on and off but it is no longer used as I changed my mind on the design and this was ditched. I mean it will still work as it's physically still there for now but I decided to not bother switching the lights on/off as they have their own light sensors anyway.

A PIR (passive infrared) detector is fitted to the front of the camera and a USB port on the rear (this is only 'USB' in the sense that it uses a USB port and cable to transmit power and signals to/from a relay inside the camera that enables or disables the IR-cut filter auto-switching and the PIR). This is why the relay in the power box is no longer used as the USB cable only has 4 wires, 2 for power and then 2 I could use as signal wires. The extra 'USB' port on the Pi is wired directly to the GPIO (5v, GND, GPIO 23 and 24). These 2 boxes for the PIR and USB port were additional to the camera's normal casing as the camera was to be used for a different project originally. As I didn't see the pointing in ditching the casing and printing another the 2 extra modules were instead retro-fitted to it.

All cases are 3D printed from my own designs (well admittedly the Pi one is a replica of a case I bought before) with the camera case being what I would call semi-waterproof. It should survive normal rainfall without too much water getting in but there are also internal drain holes and baffles that should stop it ever getting to the electronics. The window in the top of the case is to allow light to reach the light sensor that decides if to switch to IR mode or not. This switching can be disabled, whilst in either mode, by controlling the relay. This means if I still want to shoot at night, but not in IR, I can do so.

The camera is controlled with raspiCam and thanks to the new added controls (GPIO lockout and the ability to enable/disable stopping of a recording that is in progress from the GPIO) the PIR now can be set to trigger both photos whenever motion is detected and also 1 minute videos upon motion. The PIRs signal triggers pin 23 and raspCam is configured accordingly so it then just works like a physical switch you would press.

Originally using the PIR was an issue for video triggering as the PIR would start a video recording and then promptly stop it when the PIR triggered again. This could be configured out on the PIR itself but that then gave a big delay between photos being taken so after posting about possible solutions to this the two features above were added, very quickly (thanks!), and now it is 100% perfect.

There's still a lot of experimenting with settings to do to get the best exposure etc but the whole setup works really well now and raspiCam is perfectly controlling it. 

Additionally the Pi is setup to use my WiFi when I am within range but will also create it's own AP if not, meaning I can take it wherever and still remote to it by connecting to that self created AP.

I have another 8MP IR only camera that I may well convert over to using raspiCam and the same PIR setup soon too.

Any questions let me know. :)

**Reply by Lillifee** [reply]
Thank you for taking the time to create the post and explain how you utilized it. I really appreciate your effort!!!
The camera housing with the PIR sensor looks amazing, like it came straight off the shelf. 
I wasn't aware of the CSI to HDMI adapter - that's a great solution for extending the fragile cable. Thanks for sharing.
I have been considering adding a feature to create an access point if the WIFI is not available. Hopefully, I will find the time to add this feature in the future.

By the way, great picture! Although the house looks a bit small for you. ;) 👍
Thanks a lot!!!

**Reply by yq3w** [reply]
May I ask if you
How does Raspberry Pi achieve WiFi and AP hotspot switching?

**Reply by Lillifee** [reply]
Hey @yq3w,

Initially, I considered adding a feature to the raspiCam that would enable it to act as an access point if your Wi-Fi isn't within range. However, there are already existing solutions available, such as:

[AutoHotspot-Installer](https://github.com/RaspberryConnect/AutoHotspot-Installer)
[Raspberry Pi Auto WiFi Hotspot Switch Internet](https://www.raspberryconnect.com/projects/65-raspberrypi-hotspot-accesspoints/157-raspberry-pi-auto-wifi-hotspot-switch-internet)
[auto-hotspot](https://github.com/0unknwn/auto-hotspot)
[auto hotspot script](https://gist.github.com/hekigan/3e79ba728065fd88b3d6a6a88eaf2942)

I haven't used the AP feature recently, so I'm unsure which of these solutions is still actively maintained and functional.
